oauth2/oauth2_connector.bal

```diff
@@ -227,29 +228,35 @@ returns (boolean) | http:HttpConnectorError {
 }
 
 function <OAuth2Connector oAuth2Connector> getAccessTokenFromRefreshToken (http:Request request)
-                                                        returns (string) | http:HttpConnectorError {
+returns (string) | http:HttpConnectorError {
     http:HttpClient refreshTokenClient = http:createHttpClient(oAuth2Connector.refreshTokenEP,
                                                                oAuth2Connector.clientConfig);
     http:Request refreshTokenRequest = {};
     http:Response httpRefreshTokenResponse = {};
     http:HttpConnectorError connectorError = {};
     boolean useUriParams = oAuth2Connector.useUriParams;
     string accessTokenFromRefreshTokenReq = oAuth2Connector.refreshTokenPath;
-    string requestParams = "refresh_token=" + oAuth2Connector.refreshToken
-                                            + "&grant_type=refresh_token&client_secret=" + oAuth2Connector.clientSecret
-                                            + "&client_id=" + oAuth2Connector.clientId;
-    if(useUriParams) {
+
+    if(!useUriParams) {
+        string clientIdSecret = oAuth2Connector.clientId + ":" + oAuth2Connector.clientSecret;
```

Comment:
useUriParams is used to specify whether send the parameters in the request body or as uri params. Client id and client secret can be sent as the request parameters or as a basic auth header. This will break the first case. Can you please introduce another parameter and use it.

Reply:
By setting useUriParams = true the first case can still be accomplished.

Reply:
If the useUriParams = true, then these request parameters will be set as uri params. But what if we need to send all these parameters in the request body. What you are doing is, setting the client credentials as basic auth header and other params in the body which is a different case.